spring-data-cassandra/src/main/java/org/springframework/data/cassandra/config/CassandraSessionFactoryBean.java

```diff
@@ -13,109 +13,153 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
 package org.springframework.data.cassandra.config;
 
+import static org.springframework.cassandra.core.cql.CqlIdentifier.cqlId;
+
 import java.util.Collection;
 
 import org.springframework.cassandra.config.CassandraCqlSessionFactoryBean;
 import org.springframework.data.cassandra.convert.CassandraConverter;
+import org.springframework.data.cassandra.core.CassandraAdminOperations;
 import org.springframework.data.cassandra.core.CassandraAdminTemplate;
 import org.springframework.data.cassandra.mapping.CassandraMappingContext;
 import org.springframework.data.cassandra.mapping.CassandraPersistentEntity;
 import org.springframework.util.Assert;
 
 import com.datastax.driver.core.KeyspaceMetadata;
 import com.datastax.driver.core.Metadata;
+import com.datastax.driver.core.Session;
 import com.datastax.driver.core.TableMetadata;
 
-import static org.springframework.cassandra.core.cql.CqlIdentifier.cqlId;
-
+/**
+ * Factory to create and configure a Cassandra {@link com.datastax.driver.core.Session} with support
+ * for executing CQL and initializing the database schema (a.k.a. keyspace).
+ *
+ * @author Mathew Adams
+ * @author David Webb
+ * @author John Blum
+ * @see com.datastax.driver.core.KeyspaceMetadata
+ * @see com.datastax.driver.core.TableMetadata
+ */
 public class CassandraSessionFactoryBean extends CassandraCqlSessionFactoryBean {
 
-	protected SchemaAction schemaAction = SchemaAction.NONE;
-	protected CassandraAdminTemplate admin;
-	protected CassandraConverter converter;
-	protected CassandraMappingContext mappingContext;
+	protected static final boolean DEFAULT_CREATE_IF_NOT_EXISTS = false;
+	protected static final boolean DEFAULT_DROP_TABLES = false;
+	protected static final boolean DEFAULT_DROP_UNUSED_TABLES = false;
+
+	private CassandraAdminOperations admin;
+
+	private CassandraConverter converter;
+
+	private SchemaAction schemaAction = SchemaAction.NONE;
 
 	@Override
 	public void afterPropertiesSet() throws Exception {
-
 		super.afterPropertiesSet();
 
-		Assert.notNull(converter);
+		Assert.state(converter != null, "Converter must not be null");
```

Comment:
I think the Exception is important in this case (well, in most cases) as it communicates either the state of the object/system (`IllegalStateException`), the validity of the values passed in (`IllegalArgumentException`) or whether the programmer (caller) made a mistake using the API (`NullPointerException`, a programmatical `Exception`).

In this, case the `IllegalStateException` represents that the object is not in a proper state to finish post initialization activities, therefore, the `IllegalStateException` is the appropriate `Exception`.

However, what is probably not appropriate is the message.  The `Exception` message should rather read...

> Converter was not properly initialized

Will change.

Side note...

I do think there is an excessive number of inappropriate `Assert.notNull(object)`, `Assert.isTrue(condition)`, etc in the Spring Data codebase that should be including a **message**.

One of the reasons for this is, the Spring `Assert` utility class uses a generic message when one is not given, for example... `[Assert.notNull(object)](https://github.com/spring-projects/spring-framework/blob/v4.2.6.RELEASE/spring-core/src/main/java/org/springframework/util/Assert.java#L126)`.

The problem is, if a constructor or method with multiple arguments (for [example](https://github.com/spring-projects/spring-data-commons/blob/master/src/main/java/org/springframework/data/mapping/model/AbstractPersistentProperty.java#L61-L62)) asserts not null on those arguments without a message, such as...

```
public Object someMethod(Object argument1, Object argument2, ...) {
    Assert.notNull(argument1);
    Assert.notNull(argument2);

    ...
}
```

Then, the caller has no idea which argument was null, given the user no context and helpful debug information.

Food for thought.